spec.emu

```diff
@@ -143,25 +177,23 @@ contributors: Ujjwal Sharma, Younies Mahmoud
         1. Let _style_ be ? GetOption(_options_, `"style"`, `"string"`, &laquo; `"long"`, `"short"`, `"narrow"`, `"digital"` &raquo;, `"long"`).
         1. Set _durationFormat_.[[Style]] to _style_.
         1. Set _durationFormat_.[[DataLocale]] to _r_.[[dataLocale]].
-        1. Let _fields_ be ? Get(_options_, `"fields"`).
-        1. If _fields_ is *undefined*, then
-          1. Let _list_ be a new empty List.
-          1. For each row in <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
-            1. Let _field_ be the Property value.
-            1. Append _field_ as the last element of _list_.
-          1. Set _durationFormat_.[[Fields]] to _list_.
-        1. Else if IsArray(_fields_) is *true*, then
-          1. Set _fields_ to ? CreateListFromArrayLike(_fields_).
-          1. For each element _field_ in _fields_, in List order, do
-            1. If _field_ isn't in <emu-xref href="#table-duration-components"></emu-xref>, then
-              1. Throw a *RangeError* exception.
-          1. Set _durationFormat_.[[Fields]] to _fields_.
-        1. Else,
-          1. Throw a *TypeError* exception.
+        1. Let _smallestUnit_ be ? ToSmallestTemporalDurationUnit(_options_, &laquo; &raquo; `undefined`).
+        1. Set _durationFormat_.[[SmallestUnit]] to _smallestUnit_.
+        1. Let _largestUnit_ be ? ToTemporalDurationUnit(_options_, `"largestUnit"`, `undefined`).
+        1. If _smallestUnit_ is listed above _largestUnit_ in <emu-xref href="#table-duration-components"></emu-xref>,
```

Comment:
_smallestUnit_ (or maybe also _largestUnit_ , depending what ToTemporalDurationUnit do) could be undefined. So maybe this IF block should be inside a "if _smallestUnit_ is not *undefined*" block to make it clear.